sdk/python/tests/conftest.py

```diff
@@ -52,18 +54,27 @@ def pytest_addoption(parser):
     )
 
 
-def pytest_collection_modifyitems(config, items):
+def pytest_collection_modifyitems(config, items: List[Item]):
     should_run_integration = config.getoption("--integration") is True
     should_run_benchmark = config.getoption("--benchmark") is True
-    skip_integration = pytest.mark.skip(
-        reason="not running tests with external dependencies"
-    )
-    skip_benchmark = pytest.mark.skip(reason="not running benchmarks")
-    for item in items:
-        if "integration" in item.keywords and not should_run_integration:
-            item.add_marker(skip_integration)
-        if "benchmark" in item.keywords and not should_run_benchmark:
-            item.add_marker(skip_benchmark)
+
+    integration_tests = [t for t in items if "integration" in t.keywords]
+    if not should_run_integration:
+        for t in integration_tests:
+            items.remove(t)
+    else:
```

Comment:
Seems like these are exclusive? 

If the user does should_run_integration and should_run_benchmark, it ends up only running benchmark tests.

I'd expect this more to be

if not should_run_integration, remove integration tests
if not should_run_benchmarks, remove benchmarks
done

Reply:
I changed the current behaviour to be exclusive.

if user specifies `--benchmark` and `--integration`, it runs tests marked with both. If the user specified only one of them, it runs tests with only those marks.

i prefer this behaviour since otherwise we run unit tests during all our integration tests - which seems like a duplication and a waste. It also is more straightforward to reason about, IMO.

Reply:
The reason why `integration` and `benchmark` tests are orthogonal is because it's possible for an integration test to need access to resources on aws/gcp, which requires the `integration` mark so that it can be set up correctly.